File: redome-front/src/app/shared/dto/plano.workup.internacional.dto.ts
import { BuildForm } from "../buildform/build.form";
import { PlanoWorkupDTO } from "./plano.workup.dto";
import { PlanoWorkupNacionalDTO } from "./plano.workup.nacional.dto";
import { DataUtil } from "../util/data.util";
import { FormGroup } from "@angular/forms";

export class PlanoWorkupInternacionalDTO extends PlanoWorkupDTO {

	private _observacaoPlanoWorkup: String;

    public static parse(formCentro: FormGroup): PlanoWorkupInternacionalDTO{

        let planoWorkupInternacionalDTO = new PlanoWorkupInternacionalDTO();
        planoWorkupInternacionalDTO.dataExame = DataUtil.toDate(formCentro.get("dataExame").value);
        planoWorkupInternacionalDTO.dataResultado = DataUtil.toDate(formCentro.get("dataResultado").value);
        planoWorkupInternacionalDTO.dataColeta = DataUtil.toDate(formCentro.get("dataColeta").value);
        planoWorkupInternacionalDTO.observacaoPlanoWorkup = formCentro.get("observacao").value;

        return planoWorkupInternacionalDTO;
    }



    /**
     * Getter observacaoPlanoWorkup
     * @return {String}
     */
	public get observacaoPlanoWorkup(): String {
		return this._observacaoPlanoWorkup;
	}

    /**
     * Setter observacaoPlanoWorkup
     * @param {String} value
     */
	public set observacaoPlanoWorkup(value: String) {
		this._observacaoPlanoWorkup = value;
	}


    public jsonToEntity(res: any) {
        throw new Error("Method not implemented.");
    }

}

File: redome-workup/src/main/java/br/org/cancer/redome/workup/dto/PlanoWorkupInternacionalDTO.java
package br.org.cancer.redome.workup.dto;

import java.io.Serializable;
import java.time.LocalDate;

import lombok.Builder;
import lombok.EqualsAndHashCode;
import lombok.Getter;
import lombok.NoArgsConstructor;


//@Data
@EqualsAndHashCode(callSuper = true)
@NoArgsConstructor
@Getter
public class PlanoWorkupInternacionalDTO extends PlanoWorkupDTO implements Serializable {

	private static final long serialVersionUID = 8814050491745118764L;
	
	private String observacaoPlanoWorkup;
	
	/*
	 * public PlanoWorkupNacionalDTO() { super(); }
	 */
	
	@Builder
	public PlanoWorkupInternacionalDTO(LocalDate dataExame, LocalDate dataResultado,  LocalDate dataColeta, Long idCentroTransplante, String observacaoPlanoWorkup ) {
		super(dataExame, dataResultado, dataColeta, idCentroTransplante);
		this.observacaoPlanoWorkup = observacaoPlanoWorkup;
	}
	
	
	
}
